backend/devices/bitbox02/keystore.go

```diff
@@ -396,16 +364,21 @@ func (keystore *keystore) signBTCTransaction(btcProposedTx *btc.ProposedTransact
 		default:
 			return errp.Newf("unsupported output type: %v", address)
 		}
-		changeAddress := btcProposedTx.TXProposal.ChangeAddress
-		isChange := changeAddress != nil && bytes.Equal(
-			changeAddress.PubkeyScript(),
-			txOut.PkScript,
-		)
+		outputAddress := btcProposedTx.GetAddress(blockchain.NewScriptHashHex(txOut.PkScript))
+		isChange := outputAddress != nil && outputAddress.IsChange()
```

Comment:
could actually also use `btcProposedTx.TXProposal.ChangeAddress` to check if it is change, maybe that would be better, but this way we could delete `ChangeAddress` from the struct :shrug: either is okay I think.

Reply:
switched back to using ChangeAddress for simplicity